sound/soc/intel/common/soc-acpi-intel-rpl-match.c

```diff
@@ -407,6 +407,13 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_rpl_machines[] = {
 		.quirk_data = &rpl_max98373_amp,
 		.sof_tplg_filename = "sof-rpl-max98373-nau8825.tplg",
 	},
+	{
+		.id = "10508825",
+		.drv_name = "rpl_mx98360a_nau8825",
```

Comment:
Should this match the board id name? I.e. "rpl_mx98360a_8825"

Reply:
I would think this is named mx98360a due to length limitation as we do in the adl [variant](https://github.com/thesofproject/linux/blob/61a9c60e31ba6febebf2b2ab6a0d8bece8568bd9/sound/soc/intel/boards/sof_nau8825.c#L523) as well.

Reply:
Right, but the drv_name is ["adl_mx98360a_8825"](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-adl-match.c#L538) which matches [.name = "adl_mx98360a_8825"](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_nau8825.c#L524).
IMHO, it won't find the machine driver if .drv_name doesn't match device id name.

Reply:
Oops, thank you for catching this. Let me get this fixed.